pwiz_tools/Skyline/Model/AuditLog/AuditLogEntry.cs

```diff
@@ -265,21 +320,20 @@ public static bool ReadFromXmlTextReader(XmlTextReader reader, out string logged
                 .ChangeActualHash(result.CalculateRootHash())
                 .ChangeSkylHash(Hash.FromBase64(rootHash));
 
+            // yell if audit log entry hash does not match.
             // If the docFormat is null, this is an old audit log and there won't be any entry hashes.
             // We can't just always ignore non-existent hashes, otherwise people could just delete the hash elements
             // and get Skyline to successfully load the audit log
-            if (docFormat != null && !result.RootHash.SkylAndActualHashesEqual())
+            var modifiedEntries =
+                result.AuditLogEntries.Enumerate().Where(entry => !entry.Hash.SkylAndActualHashesEqual());
```

Comment:
You should call "ToArray()" or "ToList()" here so you don't get the Resharper warning about enumerating the collection more than once.